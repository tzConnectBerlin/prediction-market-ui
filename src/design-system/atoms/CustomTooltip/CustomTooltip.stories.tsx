import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { CustomTooltip, CustomTooltipProps } from './CustomTooltip';

export default {
  title: 'Atom/CustomTooltip',
  component: CustomTooltip,
} as Meta;

const Template: Story<CustomTooltipProps> = (args) => <CustomTooltip {...args} />;

export const Primary = Template.bind({});
Primary.args = {};

export const OpenTooltip = Template.bind({});
OpenTooltip.args = {
  open: true,
};

export const CustomColor = Template.bind({});
CustomColor.args = {
  color: 'red',
};
