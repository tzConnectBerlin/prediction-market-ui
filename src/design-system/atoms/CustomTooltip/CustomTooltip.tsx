import styled from '@emotion/styled';
import { IconButton, useTheme } from '@mui/material';
import * as React from 'react';
import { IoMdInformation } from 'react-icons/io';
import { CloseIcon, CircleBackground } from '../CloseIcon';

export interface CustomTooltipProps {
  open?: boolean;
  color?: string;
  onClick?: () => void | Promise<void>;
}

const StyledIconButton = styled(IconButton)`
  padding: 0;
  &:hover {
    background-color: rgba(0, 0, 0, 0);
  }
`;

const defaultOpen = false;

export const CustomTooltip: React.FC<CustomTooltipProps> = ({
  open = defaultOpen,
  color,
  onClick,
}) => {
  const theme = useTheme();
  const colorToUse = color ?? theme.palette.primary.main;
  return (
    <StyledIconButton disableFocusRipple disableRipple disableTouchRipple onClick={onClick}>
      {open && <CloseIcon color={colorToUse} />}
      {!open && (
        <CircleBackground>
          <IoMdInformation fill={colorToUse} size="0.7em" />
        </CircleBackground>
      )}
    </StyledIconButton>
  );
};
