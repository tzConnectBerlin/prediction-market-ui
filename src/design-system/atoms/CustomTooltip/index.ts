export { CustomTooltip, CustomTooltip as default } from './CustomTooltip';
export type { CustomTooltipProps } from './CustomTooltip';
