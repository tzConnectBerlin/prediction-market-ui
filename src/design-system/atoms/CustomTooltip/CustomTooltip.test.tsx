import renderer from 'react-test-renderer';
import { CustomTooltip } from './CustomTooltip';

describe('Snapshot testing CustomTooltip Component', () => {
  it('renders correctly with default size and color', () => {
    const tooltip = renderer.create(<CustomTooltip />).toJSON();
    expect(tooltip).toMatchSnapshot();
  });

  it('renders correctly with custom color', () => {
    const tooltip = renderer.create(<CustomTooltip color="red" />).toJSON();
    expect(tooltip).toMatchSnapshot();
  });

  it('renders correctly open tooltip', () => {
    const tooltip = renderer.create(<CustomTooltip open />).toJSON();
    expect(tooltip).toMatchSnapshot();
  });
});
