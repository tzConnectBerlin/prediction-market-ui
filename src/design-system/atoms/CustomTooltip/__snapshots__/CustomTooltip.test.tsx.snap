// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing CustomTooltip Component renders correctly open tooltip 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 0;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0);
}

.emotion-1 {
  border-radius: 50%;
  background: rgba(1, 102, 255, 0.16);
  height: 0.7em;
  width: 0.7em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-0"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="emotion-1"
  >
    <svg
      fill="#1976d2"
      height="0.7em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="0.7em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
      />
    </svg>
  </div>
</button>
`;

exports[`Snapshot testing CustomTooltip Component renders correctly with custom color 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 0;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0);
}

.emotion-1 {
  border-radius: 50%;
  background: rgba(1, 102, 255, 0.16);
  height: 0.7em;
  width: 0.7em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-0"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="emotion-1"
  >
    <svg
      fill="red"
      height="0.7em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="0.7em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M232 235h48v137h-48zM232 140h48v48h-48z"
      />
    </svg>
  </div>
</button>
`;

exports[`Snapshot testing CustomTooltip Component renders correctly with default size and color 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 0;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0);
}

.emotion-1 {
  border-radius: 50%;
  background: rgba(1, 102, 255, 0.16);
  height: 0.7em;
  width: 0.7em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-0"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="emotion-1"
  >
    <svg
      fill="#1976d2"
      height="0.7em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="0.7em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M232 235h48v137h-48zM232 140h48v48h-48z"
      />
    </svg>
  </div>
</button>
`;
